src/main/java/org/opensearch/neuralsearch/processor/AgenticQueryTranslatorProcessor.java

```diff
@@ -134,12 +161,24 @@ private void executeAgentAsync(
 
                 requestListener.onResponse(request);
             } catch (IOException e) {
-                log.error("Failed to parse agent response", e);
-                requestListener.onFailure(e);
+                String errorMessage = String.format(
+                    Locale.ROOT,
+                    "Agentic search failed - Parse error - Agent ID: [%s], Query: [%s], Error: [%s]",
```

Comment:
it's is too verbose for the error message? We can lower risk by truncating or hashing query text for user facing message, and keep full content as part of log message.

Reply:
removed query from this